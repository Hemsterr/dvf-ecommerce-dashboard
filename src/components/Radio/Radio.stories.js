// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Radio from './index'

storiesOf('Radio', module)
  .add('Default', () => <Radio value="test" label="color thread" />)
  .add('checked', () => <Radio label="color thread" isChecked />)
  .add('disabled', () => <Radio label="color thread" disabled />)
  .add('checked and disabled', () => (
    <Radio label="color thread" disabled isChecked />
  ))
