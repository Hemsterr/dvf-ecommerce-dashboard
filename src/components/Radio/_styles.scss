.radio {
  color: $black;
  letter-spacing: pxToRem(1.67px);
  line-height: pxToRem(30px);
  cursor: pointer;

  font: {
    family: $font-base-semi-bold;
    size: pxToRem(10px);
  }

  &__img {
    position: relative;
    padding-left: pxToRem(20px);
    margin-right: pxToRem(10px);
    top: pxToRem(7px);

    &::before {
      @include absolute-position(auto, auto, auto, 0);

      content: '';
      width: pxToRem(16px);
      height: pxToRem(16px);
      box-sizing: border-box;
      border-radius: 50%;
      border: pxToRem(1px) solid $black;
      background-color: $white;
    }
  }

  &__input {
    display: none;

    &:checked + .radio__img::after {
      @include absolute-position(pxToRem(-4px), auto, auto, pxToRem(4px));
      content: '';
      width: pxToRem(8px);
      height: pxToRem(8px);
      box-sizing: border-box;
      border-radius: 50%;
      border: pxToRem(1px) solid $black;
      background-color: $big-stone;
    }
  }
}
