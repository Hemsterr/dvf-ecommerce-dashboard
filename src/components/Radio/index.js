// @flow
import React from 'react'

type Props = {
  value: string,
  label: string,
  className: string,
  isChecked?: boolean,
  disabled: boolean,
  setChecked: Function,
}

class Radio extends React.PureComponent<Props> {
  static defaultProps = {
    value: '',
    label: '',
    className: '',
    isChecked: false,
    disabled: false,
    setChecked: () => {},
  }

  render() {
    const {
      label,
      value,
      className,
      isChecked,
      disabled,
      setChecked,
    } = this.props

    return (
      <label className={`radio ${className}`}>
        <input
          type="radio"
          className="radio__input"
          value={value}
          checked={isChecked}
          disabled={disabled}
          onChange={() => setChecked(value)}
        />
        <span className="radio__img" />
        {label}
      </label>
    )
  }
}

export default Radio
