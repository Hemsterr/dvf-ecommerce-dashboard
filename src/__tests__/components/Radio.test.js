// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import Radio from '../../components/Radio'

const mockProps = {
  value: 'Color Thread',
  label: 'color thread',
  isChecked: true,
  disabled: true,
}

const wrapper = (props = {}) => {
  const component = shallow(<Radio {...props} />)

  return component
}

describe('Radio component', () => {
  const wrapperWithoutProps = wrapper()
  const wrapperWithProps = wrapper(mockProps)

  // Snapshot
  test('should render correctly Radio component default props', () => {
    expect(shallowToJson(wrapperWithoutProps)).toMatchSnapshot()
  })

  // Testing props
  describe('Handle with props', () => {
    test('should is rendered default props', () => {
      expect(wrapperWithoutProps.find('input').props().value).toBeFalsy()
      expect(wrapperWithoutProps.find('input').props().checked).toBeFalsy()
      expect(wrapperWithoutProps.find('input').props().disabled).toBeFalsy()
    })

    test('should is rendered with a custom Props', () => {
      expect(wrapperWithProps.find('input').props().value).toMatch(
        mockProps.value
      )
      expect(wrapperWithProps.find('input').props().checked).toBe(true)
      expect(wrapperWithProps.find('input').props().disabled).toBe(true)
      expect(wrapperWithProps.find('label').props().children).toContain(
        mockProps.label
      )
    })
  })

  // Testing events
  describe('Checking events', () => {
    test('check the on Change callback ', () => {
      const input = wrapperWithoutProps.find('input')
      input.simulate('change')
    })
  })
})
