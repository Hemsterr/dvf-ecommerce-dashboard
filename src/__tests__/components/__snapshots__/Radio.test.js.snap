// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component should render correctly Radio component default props 1`] = `
<label
  className="radio "
>
  <input
    checked={false}
    className="radio__input"
    disabled={false}
    onChange={[Function]}
    type="radio"
    value=""
  />
  <span
    className="radio__img"
  />
</label>
`;
